dm/master/election.go

```diff
@@ -127,20 +127,26 @@ func (s *Server) closeLeaderClient() {
 	}
 }
 
-func (s *Server) isLeaderAndNeedForward() (isLeader bool, needForward bool) {
-	retry := 10
-	wait := 100 * time.Millisecond
-	for ; retry > 0; retry-- {
-		if s.leader.Get() == oneselfStartingLeader {
-			time.Sleep(wait)
-		} else {
-			break
+func (s *Server) isLeaderAndNeedForward(ctx context.Context) (isLeader bool, needForward bool) {
+	// maybe in `startLeaderComponent`, will wait for a short time
+	if s.leader.Get() == oneselfStartingLeader {
+		retry := 10
+		ticker := time.NewTicker(100 * time.Millisecond)
+		defer ticker.Stop()
+
+		for s.leader.Get() == oneselfStartingLeader {
+			if retry == 0 {
+				return false, false
```

Comment:
How about adding a log for this case?

Reply:
done in https://github.com/pingcap/dm/pull/1118/commits/020197916448fd1cc2d0a71df6d18b63995f63c2